include/pressio/ops/epetra/ops_dot.hpp

```diff
@@ -51,26 +51,44 @@
 
 namespace pressio{ namespace ops{
 
-template <typename vec_type>
+template <typename T1, typename T2, typename DotResult>
 ::pressio::mpl::enable_if_t<
-  ::pressio::is_vector_epetra<vec_type>::value
+  // TPL/container specific
+     ::pressio::is_vector_epetra<T1>::value
+  && ::pressio::is_vector_epetra<T2>::value
+  // scalar compatibility
+  && ::pressio::all_have_traits_and_same_scalar<T1, T2>::value
+  && (std::is_floating_point<typename ::pressio::Traits<T1>::scalar_type>::value
+   || std::is_integral<typename ::pressio::Traits<T1>::scalar_type>::value)
+  && std::is_convertible<
+    typename ::pressio::Traits<T1>::scalar_type,
+    DotResult>::value
   >
```

Comment:
now that i think about it more carefully, all these checks on the scalar type are not necessary since Epetra is NOT a class template, so the scalar is always double... i can merge this for now but later maybe we can fix this